torch/_inductor/codegen/cpp.py

```diff
@@ -3801,6 +3830,47 @@ def is_lowp_fp_store(node: torch.fx.Node):
                     # Hence, we remove the first to_type.
                     to_lowp_fp_legalized_nodes.append(_node)
                     _node.args = (ops, x, torch.float)
+                elif _node.target == "to_dtype_bitcast":
+                    (ops, value_var, dtype, src_dtype) = _node.args
+
+                    # to_dtype_bitcast act as a lowp fp sink:
+                    # c10::bit_cast requires the source and target have the bitwidth. Because the input tensor's dtype
+                    # could be promoted, e.g. from float16 to float, we have to cast the tensor to its original source
+                    # dtype before invoking bit_cast.
+                    if src_dtype in DTYPE_LOWP_FP:
+                        # No need to promote to float if it is a user of a lowp fp sources
+                        # which are all directly fed to ops that accepts lowp fp input
+                        if not (
+                            is_lowp_fp_source(value_var, src_dtype)
+                            and all(
+                                is_lowp_fp_sink(user, src_dtype)
+                                for user in value_var.users
+                            )
```

Comment:
Can we factor out a function to avoid code dup?

Reply:
```python
            def is_lowp_fp_source_no_promote(node: torch.fx.Node, dt: torch.dtype):
                """Check if the node is a lowp fp sources which are all directly fed to ops that accepts lowp fp input
                thus no need to promote to float
                """
                return is_lowp_fp_source(node, dt) and all(
                    is_lowp_fp_sink(user, dt) for user in node.users
                )
```

Like this? Not sure if there is a better name....

Reply:
Added in a63deaf